packages/core/src/module-resolver.ts

```diff
@@ -542,12 +542,22 @@ export class Resolver {
   }
 
   private *componentJSCandidates(inPackageName: string) {
+    const extensions = ['.ts', '.gjs', '.gts'];
```

Comment:
vite can handle this files. webpack can also already handle ts files, for gjs/gts a loader could be added